### PR TITLE
Separating Schema label from code identifier

### PR DESCRIPTION
**Pull request by hand-dot**
ref: https://github.com/pdfme/pdfme/issues/303

CleanShot 2023-11-01 at 11 29 59@2x

### CODE REVIEW

File: packages/common/__tests__/helper.test.ts [hunk 1]
```diff
@@ -219,7 +219,7 @@ describe('checkPlugins test', () => {
       pdf: async () => {},
       ui: async () => {},
       propPanel: {
-        propPanelSchema: {},
+        schema: {},
```
Comment (hand-dot) [inline packages/common/__tests__/helper.test.ts:222]: This fix is not directly related to this PR.

File: packages/common/src/helper.ts [hunk 2]
```diff
@@ -142,14 +142,11 @@ export const checkPlugins = (arg: { plugins: Plugins; template: Template }) => {
   } = arg;
   const allSchemaTypes = uniq(schemas.map((s) => Object.values(s).map((v) => v.type)).flat());
 
-  // text and image are builtin schema types so they are not included in plugins.
-  const exceptBuiltinSchemaTypes = allSchemaTypes.filter((t) => t !== 'text' && t !== 'image');
+  const pluginsSchemaTypes = Object.values(plugins).map((p) => p?.propPanel.defaultSchema.type);
 
-  const pluginsSchemaTypes = Object.keys(plugins);
```
Comment (hand-dot) [inline packages/common/src/helper.ts:148]: Stopped using the 'plugins' key.

File: packages/generator/src/generate.ts [hunk 3]
```diff
@@ -62,7 +55,9 @@ const generate = async (props: GenerateProps) => {
           continue;
         }
 
-        const render = rendererRegistry[schema.type];
+        const render = Object.values(plugins).find(
+          (plugin) => plugin.propPanel.defaultSchema.type === schema.type
```
Comment (hand-dot) [inline packages/generator/src/generate.ts:59]: There may be users who are only using the generator, but accessing `plugin.propPanel.defaultSchema.type` feels a bit awkward. Should it be accessible through `plugin.type` instead?

File: packages/generator/src/types.ts [hunk 4]
```diff
@@ -1,11 +1,3 @@
-import type { Schema, PDFRenderProps } from '@pdfme/common';
```
Comment (hand-dot) [inline packages/generator/src/types.ts:1]: We just use plugin.

This was a welcome side effect.
Thank you for making this proposal.

File: packages/generator/src/generate.ts (hunk 3 above)
Comment (hand-dot) [reply packages/generator/src/generate.ts:59]: I might be overthinking this. If it doesn't feel awkward as is, I'd like to keep it that way.

**Review by peteward — COMMENTED** [review]
this is looking a lot cleaner :+1:
